ZThread.cs

```diff
@@ -113,7 +114,8 @@ public virtual void Close()
 
 			Stop();
 
-			_thread?.Join();
+			if (_thread == null) return;
```

Comment:
I guess you changed this because some compiler does not support the new syntax introduced with C# 6.0. Which language versions are supported? This should be documented, and, if possible, the CI should also enforce this.

Reply:
clearly stated in `ZeroMQ.*.csproj` `<TargetFrameworkVersion>v4.0`

Reply:
That's something different. The (source) language version is independent from the target framework version: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/langversion-compiler-option

Reply:
I don't know man. I do believe this SHOULD work on NET 4, and doesn't need because of one or two operators, to run just on .NET 4.5

Reply:
As I said, you are mixing things up here. .NET 4 and .NET 4.5 are different target frameworks. _thread?.Join(); is compiled to CIL by a C# compiler, which understands a certain C# version. 

The Microsoft C# compiler can be told to compile some C# code restricted to only an older C# version. While the C# compiler from Visual Studio 2017 supports C# 7.0, it can be run in a mode where it only has support for C# 5.0 features enabled?

But regardless of the C# version, the target framework can still be .NET 4.

Which versions of which compiler should be supported in your opinion?

Reply:
Opened #125 for this issue.